cosmetics-web/app/analyzers/read_data_analyzer.rb

```
@@ -67,12 +69,18 @@ def create_notification_from_file
     rescue StandardError
       Rails.logger.error "StandardError"
       @notification_file.update(upload_error: :unknown_error)
+    rescue UnknownError
+      @notification_file.update(upload_error: :unknown_error)
     end
   end
 
   def get_product_xml_file
     download_blob_to_tempfile do |zip_file|
       Zip::File.open(zip_file.path) do |files|
+        if invalid_static_files(files)
+          raise UnknownError, "UnknownError - a different static file was detected!"
```

Sorry, my last comment wasn't very clear! I think it's better to raise a specific `UnexpectedStaticFilesError` error here, but when we `rescue` it, the `upload_error` value should be `:unknown_error` (which is what you now have above).

---

Sure thanks